core/arch/arm/kernel/generic_entry_a32.S

```diff
@@ -122,8 +254,18 @@ copy_init:
 	sub	r1, r1, #1
 	bl	arm_cl1_d_invbyva
 
-	/* Enable Console */
-	bl	console_init
+#ifdef CFG_PL310
```

Comment:
Long term I'd prefer having all the #ifdef CFG_PL310 stuff in platform specific code, but for now it's OK.